From 5e0a77b8042dea0956cb515abb9ba2eda1eea2f7 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:27:03 -0400
Subject: [net] genetlink: Fix uninitialized variable in genl_validate_assign_mc_groups()

Message-id: <4a2ea6d760cae6991c71f0e977720c172c8dea50.1410528958.git.jbenc@redhat.com>
Patchwork-id: 93979
O-Subject: [RHEL7.1 net PATCH 21/23] genetlink: Fix uninitialized variable in genl_validate_assign_mc_groups()
Bugzilla: 1140656
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1140656
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 0f0e2159c0c101426b705d70f43b9f423d51c869
Author: Geert Uytterhoeven <geert@linux-m68k.org>
Date:   Sat Nov 23 13:01:50 2013 +0100

    genetlink: Fix uninitialized variable in genl_validate_assign_mc_groups()

    net/netlink/genetlink.c: In function 'genl_validate_assign_mc_groups':
    net/netlink/genetlink.c:217: warning: 'err' may be used uninitialized in this
    function

    Commit 2a94fe48f32ccf7321450a2cc07f2b724a444e5b ("genetlink: make multicast
    groups const, prevent abuse") split genl_register_mc_group() in multiple
    functions, but dropped the initialization of err.

    Initialize err to zero to fix this.

    Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/netlink/genetlink.c b/net/netlink/genetlink.c
index 2e79b96..ea38660 100644
--- a/net/netlink/genetlink.c
+++ b/net/netlink/genetlink.c
@@ -214,7 +214,7 @@ static int genl_validate_assign_mc_groups(struct genl_family *family)
 {
  int first_id;
  int n_groups = family->n_mcgrps;
- int err, i;
+ int err = 0, i;
  bool groups_allocated = false;
 
  if (!n_groups)
-- 
1.7.1